src/app/editor/form-generator/components/binary-field/binary-field.component.ts

```diff
@@ -59,30 +76,133 @@ export class BinaryFieldComponent extends BaseFieldComponent {
         this.isCompact = width <= SMALL_SCREEN_LIMIT;
     }
 
-    onFilesSelected(files: any[]): void {
+    onFilesSelected(files: File[]): void {
         const file = files[0];
-        this.api.setValue({ fileName: file.name, fileSize: file.size, mimeType: file.type, file } as BinaryField);
+        this.api.setValue(this.binaryFieldFromFile(file));
+        this.scaledTransform = {};
+        this.transformParams = undefined;
+    }
+
+    onImageLoad(): void {
+        this.loadingPreview = false;
+        if (this.binaryProperties.file && this.lastParams) {
+            this.scaledTransform = this.lastParams;
+            this.lastParams = undefined;
+        }
+    }
+
+    editImage(): void {
+        const node = this.api.getNodeValue() as MeshNode;
+        let imageUrl: string | SafeUrl;
+        const newFile = this.binaryProperties.file;
+        if (newFile) {
+            imageUrl = this.blobService.createObjectURL(this.binaryProperties.file);
+        } else {
+            imageUrl = this.apiService.project.getBinaryFileUrl(node.project.name, node.uuid, this.api.field.name);
```

Comment:
modify getBinaryFileURL to accept node.version

Reply:
done